An 118 coordinate upgrade

sucildossj opened the pull request:
- included preconf
- included precommitment
- included asset
- included asset prune
- changes on custom rpcs

sucildossj committed:
feat: fix default storage path for coordiante version 29

sucildossj committed:
refactor: including chain params changes

- include magic number changes
- include retarget diffculty change
- include bech32 prefix changes

sucildossj committed:
fix: remove gensis block check from chain params

sucildossj committed:
refactor: including auxpow chain params

sucildossj committed:
feat: including auxpow changes

sucildossj committed:
feat: including  coordinate features in bitcoin

included preconf
included precommitment
included asset

sucildossj committed:
feat: include create raw transaction support for asset

sucildossj committed:
fix: miissing import on block announcement sender

sucildossj committed:
fix: including asset max value depdency

sucildossj committed:
fix: modify version variable name changes all over script

sucildossj committed:
fix: including mempool fixes for signedblock expiry

sucildossj committed:
fix: include coin default value

sucildossj committed:
fix: including fixes for coordinate related components

sucildossj committed:
fix: including changes related to read block from disk issue

sucildossj committed:
fix: include default coin value on constructor

sucildossj committed:
fix: bitcoin block and transaction template

sucildossj committed:
fix pegin issue with bitcoin

sucildossj committed:
fix coin issue for asset transaction

sucildossj committed:
fix: signblock issue to live

sucildossj committed:
fix issue related to preconf mempool

sucildossj committed:
fix: include changes on network module

sucildossj committed:
fix issue in block storage

sucildossj committed:
fix: coin db issue and pegin issue

sucildossj committed:
fix: include test case changes to live

sucildossj committed:
fix: updating changes related to block storage and fees issue

sucildossj committed:
fix: skip auxpow validation for regtest gensis block

sucildossj committed:
fix: preconf block encode issue before sending to network

sucildossj committed:
fix: testcase fixes for amount and bip324

sucildossj committed:
fix: block filter test by modifying test vector

sucildossj committed:
fix: include bloom test fixes

sucildossj committed:
fix: remove coin test vector

sucildossj committed:
fix keyio vector to live

sucildossj committed:
fix: add new function get merkle path for normal, signed and pegin transaction

sucildossj committed:
fix: include merkle proof fix for signedblock and pegin tx

sucildossj committed:
fix rpc proof api to live

sucildossj committed:
fix: merkle block test case

jasonribble commented:
Are you using any C/C++ formatter?  @sucildossj

jasonribble commented on src/consensus/amount.h:
```diff
@@ -24,6 +25,7 @@ static constexpr CAmount COIN = 100000000;
  * for the creation of coins out of thin air modification could lead to a fork.
  * */
 static constexpr CAmount MAX_MONEY = 21000000 * COIN;
+static constexpr CAmount MAX_ASSET_OUT_VALUE = std::numeric_limits<int64_t>::max();
```
Out of curiosity, what is this about?

sucildossj replied:
we have asset that will checked against int64_t max number. it remove 21000000 restriction for asset

jasonribble replied:
Got it

jasonribble commented:
Let me know when it is ready for system level tests; and which scenarios you'd like me to cover

sucildossj commented:
> Are you using any C/C++ formatter? @sucildossj

i used clang-format before

sucildossj commented:
> Let me know when it is ready for system level tests; and which scenarios you'd like me to cover

sure i will let you know this week. Thanks

sucildossj committed:
fix: miner test test cases

jasonribble commented on src/consensus/consensus.h:
```diff
@@ -16,7 +16,7 @@ static const unsigned int MAX_BLOCK_WEIGHT = 4000000;
 static const int64_t MAX_BLOCK_SIGOPS_COST = 80000;
 /** Coinbase transaction outputs can only be spent after this number of new blocks (network rule) */
-static const int COINBASE_MATURITY = 100;
+static const int COINBASE_MATURITY = 10;
```
Why change the maturity?

sucildossj replied:
@theblackmarble  Informed to make coinbase maturity as 10. i don't know the exact reason @jasonribble

jasonribble replied:
Since pegins aren't apart of the coinbase, we don't need to have this. As long as people's coin don't get locked up for a 16 hours

sucildossj committed:
fix auxpow issue on block

sucildossj committed:
fix: blockchain test fix related to block number

sucildossj committed:
fix: including 30 testcase fixes

sucildossj committed:
fix: Merkle test changes on block transaction

sucildossj committed:
fix testcases and avoid coin checks

sucildossj committed:
fix: coinbase maturity resets to default

sucildossj committed:
doc: include clang format linting for all cpp and .h file

sucildossj committed:
fix: include changes for include asset diff in scantxoutset

sucildossj committed:
change wallet listunspent api for asset and preconf support

sucildossj committed:
fix: include reserve support on listunspent

sucildossj committed:
feat: include send pegout implementation

sucildossj committed:
updating changes related to wallet rpc

sucildossj committed:
include asset id in coutput point

sucildossj committed:
change asset id as class instead for int64_t

sucildossj committed:
remove blocktype for casset class since asset creation only belong to raw block

sucildossj committed:
include CAsset in coutpoint and updated testcase

sucildossj committed:
including asset id implementation

sucildossj committed:
included changes releated to asset outpoint changes

-- validated asset id from blockchain
-- getblocktemplate included only 256 asset creation
-- consensus validation for 256 asset creation

sucildossj committed:
refactor: disable asset creationg and transfer from mempool

sucildossj committed:
remove asset support on preconf transaction

sucildossj committed:
feat: include utxo fee logic for asset type transaction only

sucildossj committed:
refactor: fixed unit testing changes for block filter and gensis block hash changes

sucildossj committed:
include unit test changes after transaction template change

sucildossj committed:
included changes on getrawtransaction for asset id

sucildossj committed:
fix: include diffcult nan fixes on getblockheader

sucildossj committed:
including functional testcase hrp changes

sucildossj committed:
include asset doc for scantxoutset

sucildossj committed:
changed primitive class structure based on node

sucildossj committed:
updating changes for merkle root changes

sucildossj committed:
include asset transaction test file in repo

sucildossj committed:
update value on supply and controller output

sucildossj committed:
enable preconf mempool for testcase and include testcase for preconf

sucildossj committed:
Merge pull request #2 from AnduroProject/AN-170-coordinate-tx-wallet-support

An 170 coordinate tx wallet support
